app/pages/sections/catalogue-solutions/order-item/controller.test.js

```diff
@@ -101,4 +84,96 @@ describe('catalogue-solutions order-item controller', () => {
       });
     });
   });
+
+  describe('validateOrderItemForm', () => {
+    describe('when there are no validation errors', () => {
+      it('should return success as true', () => {
+        const data = {
+          quantity: '1',
+          price: '1',
+        };
+
+        const response = validateOrderItemForm({ data });
+
+        expect(response.success).toEqual(true);
+      });
+    });
+
+    describe('when there are validation errors', () => {
+      const quantityRequired = {
+        field: 'quantity',
+        id: 'quantityRequired',
+      };
+      const numericalQuantity = {
+        field: 'quantity',
+        id: 'numericQuantityRequired',
+      };
+      const priceRequired = {
+        field: 'price',
+        id: 'priceRequired',
+      };
+      const numericalPrice = {
+        field: 'price',
+        id: 'numericPriceRequired',
+      };
+
+      it('should return an array of one validation error and success as false if empty string for quantity is passed in', () => {
+        const data = {
+          quantity: '',
+          price: '1.5',
+        };
+
+        const response = validateOrderItemForm({ data });
+
+        expect(response.success).toEqual(false);
+        expect(response.errors).toEqual([quantityRequired, numericalQuantity]);
```

Comment:
should it just be one error? quantityIsRequired. Don't check the numerical until something is provided